app/models/content_change.rb

```diff
@@ -3,6 +3,14 @@
 class ContentChange < ApplicationRecord
   include SymbolizeJSON
 
+  scope :for_email, ->(email) do
+    subscription_content = SubscriptionContent
+      .where("subscription_contents.content_change_id = content_changes.id")
+      .where(email: email)
+      .exists
+    where(subscription_content)
+  end
```

Comment:
I'd maybe consider moving this out of `ContentChange` and either into `Email` or the `Service`. I'm not overly keen on giving `ContentChange` knowledge of how everything associates to it. As far as `ContentChange` is concerned it just holds some information about what's changed and the rest of the system builds its structures around it.

But... I don't really mind as it's only used for stats tracking (which is always a bit of an awkward thing). What do you think?

Reply:
Sure, I get you. Maybe it's better if I create a Query object. I had it originally in the Service but felt like something that should be tested in it's own right.

Reply:
@tuzz I've introduced this into a query class now.